woocommerce/woocommerce-ios
# [Custom Amounts M1] Custom Amount Flow

Sorry for the big PR. This is due to the refactoring to reuse the component existent in Simple Payments, including unit tests.

Closes: #10879 


## Description

With this PR we implement the flow so the merchants can add a custom amount to an order. Here we pass the amount and name to the EditableOrderViewModel, but the actual adding of the amount as a fee to the order will come in a following PR.

We already had a very similar screen for Simple Payments, because of that part of this PR was the extraction of that code to make it reusable.

As you can see, following the example of other view model references in the `EditableOrderViewModel`, we have one reference to the `AddCustomAmountViewModel` instead of recreating it multiple times. The reason is, apart from having to recreate the view model every time we need it, that when rotation the view was recreated sometimes, recreating the view model as well. Because of that, we were losing the state.

## Testing instructions


1. Go to orders
2. Tap on + to create a new order
3. Tap on + Add Custom amount
4. See that the flow is correct. After you save a custom amount, you can see in the Xcode console:

`"Adding custom amount of 12 with name Test"`

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/1864060/19624327-e4bd-4f68-9d95-41684fbdeb12

---
